packages/cadl-ranch-specs/http/payload/multipart/main.tsp

```diff
@@ -38,4 +70,180 @@ Content-Type: application/octet-stream;
   @post
   @route("/mixed-parts")
   op basic(@header contentType: "multipart/form-data", @body body: MultiPartRequest): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+Expect request:
+```
+POST /upload HTTP/1.1
+Content-Length: 428
+Content-Type: multipart/form-data; boundary=abcde12345
+
+--abcde12345
+Content-Disposition: form-data; name="id"
+Content-Type: text/plain
+
+123
+--abcde12345
+Content-Disposition: form-data; name="address"
+Content-Type: application/json
+
+{
+  "city": "X"
+}
+--abcde12345
+Content-Disposition: form-data; name="profileImage"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
+--abcde12345--
+Content-Disposition: form-data; name="previousAddresses"
+Content-Type: application/json
+
+[{
+  "city": "Y"
+},{
+  "city": "Z"
+}]
+--abcde12345
+Content-Disposition: form-data; name="pictures"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
+--abcde12345
+Content-Disposition: form-data; name="pictures"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
+--abcde12345--
+```
+""")
+  @doc("Test content-type: multipart/form-data for mixed scenarios")
+  @post
+  @route("/complex-parts")
+  op complex(@header contentType: "multipart/form-data", @body body: ComplexPartsRequest): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+Expect request:
+```
+POST /upload HTTP/1.1
+Content-Length: 428
+Content-Type: multipart/form-data; boundary=abcde12345
+
+--abcde12345
+Content-Disposition: form-data; name="address"
+Content-Type: application/json
+
+{
+  "city": "X"
+}
+--abcde12345
+Content-Disposition: form-data; name="profileImage"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
+--abcde12345--
+```
+""")
+  @doc("Test content-type: multipart/form-data for scenario contains json part and binary part ")
+  @post
+  @route("/json-part")
+  op withJsonPart(@header contentType: "multipart/form-data", @body body: JsonPartRequest): NoContentResponse;
```

Comment:
nit, I am not sure whether we need the `with`. `op jsonPart(..)` seems already ok?

Reply:
removed all prefix `with` from op name